packit/cli/builds/obs_build.py

```diff
@@ -0,0 +1,240 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import asyncio
+import logging
+import os.path
+from pathlib import Path
+import subprocess
+from tempfile import TemporaryDirectory
+
+import click
+
+from py_obs.osc import Osc
+from py_obs.person import Person, PersonRole
+from py_obs.project import (
+    Package,
+    PathEntry,
+    Project,
+    Repository,
+    send_meta,
+    upload_file_contents,
+)
+from py_obs.xml_factory import StrElementField
+
+from packit.cli.types import LocalProjectParameter
+from packit.cli.utils import cover_packit_exception, get_packit_api, iterate_packages
+from packit.config import get_context_settings, pass_config
+from packit.config.aliases import DEPRECATED_TARGET_MAP
+from packit.config.config import Config
+from packit.config.package_config import PackageConfig
+from packit.constants import (
+    PACKAGE_LONG_OPTION,
+    PACKAGE_OPTION_HELP,
+    PACKAGE_SHORT_OPTION,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def target_to_path(target: str) -> list[PathEntry]:
+    target_split = target.split("-")
+    version, arch = target_split[-2:]
+    distro = "-".join(target_split[:-2])
+
+    if distro == "fedora":
+        if version == "rawhide":
+            return [PathEntry(project="Fedora:Rawhide", repository="standard")]
+        return [
+            PathEntry(project=f"Fedora:{version}", repository="standard"),
+            PathEntry(project=f"Fedora:{version}", repository="update"),
+        ]
+
+    if distro == "epel":
+        if version == "9":
+            return [PathEntry(project=f"Fedora:EPEL:{version}", repository="stream")]
+        if version in ("8", "7"):
+            return [PathEntry(project=f"Fedora:EPEL:{version}", repository="CentOS")]
+
+    if distro == "opensuse-leap":
+        return [
+            PathEntry(project=f"openSUSE:Leap:{version}", repository="standard"),
+        ]
+
+    if distro == "opensuse":
+        if arch == "x86_64":
+            return [PathEntry(project="openSUSE:Factory", repository="snapshot")]
+        if arch in ("s390x", "aarch64", "ppc64le"):
+            postfix = {"s390x": "zSystem", "aarch64": "ARM", "ppc64le": "PowerPC"}[arch]
+            return [
+                PathEntry(project=f"openSUSE:Factory:{postfix}", repository="standard")
+            ]
+
+    raise ValueError(f"No preset available for {distro=}, {version=}, {arch=}")
+
+
+def targets_to_project(
+    targets: list[str],
+    owner: str,
+    prj_name: str,
+    description: str | None = None,
```

Comment:
Can you change `|` to `Union`. Minimum python syntax is `3.9` because of EPEL9. Also worth installing `pre-commit` locally.
```console
$ pip install -e .[dev]
$ pre-commit install
```

Reply:
Done